addons/phone_validation/tests/test_phonenumbers_patch.py

```diff
@@ -0,0 +1,28 @@
+from odoo.tests.common import TransactionCase
+from odoo.tests import tagged
+from odoo.tools.parse_version import parse_version
+import odoo.addons.phone_validation.lib.phonenumbers_patch
+
+@tagged('post_install', '-at_install')
+class TestPhonenumbersPatch(TransactionCase):
+    @classmethod
+    def test_region_CI_monkey_patch(self):
+        """Test if the  patch is apply on the good version of the lib
+        And test some phonenumbers"""
+        try:
+            import phonenumbers
+            # MONKEY PATCHING phonemetadata of Ivory Coast if phonenumbers is too old
+            if parse_version('7.6.1') <= parse_version(phonenumbers.__version__) < parse_version('8.12.32'):
+                # check that _local_load_region is set to `odoo.addons.phone_validation.lib.phonenumbers_patch._local_load_region`
+                # check that you can load a new ivory coast phone number without error
+                parsed_phonenumber_1 = phonenumbers.parse("20 25/35-51 ", region="CI", keep_raw_input=True)
+                self.assertTrue(20253551 == parsed_phonenumber_1.national_number, "The national part of the phonenumber should be 22522586")
+                self.assertTrue(255 == parsed_phonenumber_1.country_code, "The country code of Ivory Coast is 255")
+
+                parsed_phonenumber_2 = phonenumbers.parse("+225 22 52 25 86 ", region="CI", keep_raw_input=True)
+                self.assertTrue(22522586 == parsed_phonenumber_2.national_number, "The national part of the phonenumber should be 22522586")
+                self.assertTrue(255 == parsed_phonenumber_2.country_code, "The country code of Ivory Coast is 255")
+                return
+        except ImportError:
+            pass
+        self.assertFalse(hasattr(odoo.addons.phone_validation.lib.phonenumbers_patch.phonenumbers_patch, '_local_load_region'))

```

Comment:
Is it working when import fails ? Seems we should skip the whole test if import failed (easier to read test: exist if does not apply, then classic test)

Reply:
if there is an import fail the monkey patch should not be available. Skipping test explicitely be done like this:

```py
        try:
            import phonenumbers
        except ImportError:
            self.skipTest('Cannot test without phonenumbers module installed.')
        # MONKEY PATCHING phonemetadata of Ivory Coast if phonenumbers is too old
        if parse_version('7.6.1') <= parse_version(phonenumbers.__version__) < parse_version('8.12.32'):
            # ... no return needed
        else:
            self.assertFalse(hasattr(odoo.addons.phone_validation.lib.phonenumbers_patch.phonenumbers_patch, '_local_load_region'),
                "The code should not be monkey patched with phonenumbers>8.12.32.")
```